### PR TITLE
fix: allocate-tenant fails when run twice

### PR DESCRIPTION
**Pull request by prb112**
Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/PopulateParameterNames.java [hunk 1]
```diff
@@ -52,15 +54,27 @@ public PopulateParameterNames(String adminSchemaName, String schemaName, Integer
...
                             + "VALUES (%s.sv_tenant_id, ?, ?)", schemaName, adminSchemaName);
         } else {
             stmtResourceTypeInsert =
-                    String.format(
-                            "INSERT INTO %s.parameter_names (PARAMETER_NAME_ID, PARAMETER_NAME) " + "VALUES (?, ?)",
-                            schemaName);
+                    String.format("INSERT INTO %s.parameter_names (PARAMETER_NAME_ID, PARAMETER_NAME) " + "VALUES (?, ?)", schemaName);
+        }
+
+        Map<String, Integer> values = new HashMap<>();
```
Comment (punktilious) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/PopulateParameterNames.java:67]: As this is being run as maintenance task I supposed we're not going to be too worried about concurrency. Selecting first then inserting what's missing isn't thread-safe but it's probably not a big deal here.

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/PopulateParameterNames.java [hunk 2]
```diff
@@ -76,26 +90,34 @@ public void run(IDatabaseTranslator translator, Connection c) {
...
+                    // Check Error Codes.
+                    int[] codes = batch.executeBatch();
+                    int errorCodes = 0;
+                    for (int code : codes) {
+                        if (code < 0) {
+                            errorCodes++;
+                        }
+                    }
+                    if (errorCodes > 0) {
+                        String msg = "at least one of the Paramater Name/Codes are not populated [" + errorCodes + "]";
```
Comment (punktilious) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/PopulateParameterNames.java:117]: Parameter

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/PopulateParameterNames.java [hunk 3]
```diff
@@ -76,26 +90,34 @@ public void run(IDatabaseTranslator translator, Connection c) {
...
-                    batch.addBatch();
-                }
 
-                // Check Error Codes.
-                int[] codes = batch.executeBatch();
-                int errorCodes = 0;
-                for (int code : codes) {
-                    if (code < 0) {
-                        errorCodes++;
+                    if (values.containsKey(code)) {
```
Comment (punktilious) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/PopulateParameterNames.java:98]: Shouldn't this be 
```
   if (!values.containsKey(code))
```

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/PopulateResourceTypes.java [hunk 4]
```diff
@@ -66,26 +82,34 @@ public void run(IDatabaseTranslator translator, Connection c) {
...
-                    batch.addBatch();
-                }
 
-                // Check Error Codes.
-                int[] codes = batch.executeBatch();
-                int errorCodes = 0;
-                for (int code : codes) {
-                    if (code < 0) {
-                        errorCodes++;
+                    if (values.containsKey(resource)) {
```
Comment (punktilious) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/PopulateResourceTypes.java:90]: Shouldn't this be 
```
   if (!values.containsKey(code))
```

Comment (prb112) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/PopulateResourceTypes.java:90]: ack... how did this work locally... I'll fix this up

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/PopulateParameterNames.java (hunk 3 above)
Comment (prb112) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/PopulateParameterNames.java:98]: Yes! eagle eyes

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/PopulateResourceTypes.java (hunk 4 above)
Comment (prb112) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/PopulateResourceTypes.java:90]: Eagle Eyes again! thank you

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/PopulateParameterNames.java (hunk 1 above)
Comment (prb112) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/PopulateParameterNames.java:67]: Good point, I'll have to defer this part... if customers hit we can add USING RS

**Comment by prb112** [comment]
Pushed an update per code review

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/PopulateParameterNames.java (hunk 1 above)
Comment (punktilious) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/PopulateParameterNames.java:67]: Yeah, I'm not too worried about this in this context.

**Review by punktilious — APPROVED** [review]
LGTM

**Pull request merged**